pkg/provider/infisical/api/api.go

```diff
@@ -0,0 +1,255 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or impliec.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+	"github.com/external-secrets/external-secrets/pkg/metrics"
+	"github.com/external-secrets/external-secrets/pkg/provider/infisical/constants"
+)
+
+type InfisicalClient struct {
+	BaseURL *url.URL
+	client  *http.Client
+	token   string
+}
+
+type InfisicalApis interface {
+	MachineIdentityLoginViaUniversalAuth(data MachineIdentityUniversalAuthLoginRequest) (*MachineIdentityDetailsResponse, error)
+	GetSecretsV3(data GetSecretsV3Request) (map[string]string, error)
+	GetSecretByKeyV3(data GetSecretByKeyV3Request) (string, error)
+	RevokeAccessToken() error
+}
+
+const UserAgentName = "k8-external-secrets-operator"
+const errJSONSecretUnmarshal = "unable to unmarshal secret: %w"
+
+func NewAPIClient(baseURL string) (*InfisicalClient, error) {
+	baseParsedURL, err := url.Parse(baseURL)
+	if err != nil {
+		return nil, err
+	}
+
+	api := &InfisicalClient{
+		BaseURL: baseParsedURL,
+		client:  &http.Client{},
```

Comment:
Configure some timeout instead of the default empty client.
https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779